src/net_setup.c

```diff
@@ -860,9 +860,9 @@ static bool setup_myself(void) {
 
 	/* Compression */
 
-	if(get_config_int(lookup_config(config_tree, "Compression"), &myself->incompression)) {
+	if(get_config_int(lookup_config(config_tree, "Compression"), (int *) &myself->incompression)) {
```

Comment:
While in practice that will work, this is incorrect; the compiler has some freedom in choosing the underlying type of an enum, so it might not be the same size as an `int`. You could use a temporary `int` variable to read it into, and then copy the result into `myself->incompression`.